docs/Configuration/Overview.md

```diff
@@ -0,0 +1,137 @@
+---
+title: Configuration Overview
+permalink: /configuration/overview
+category: Configuration
+menuOrder: 1
+---
+
+Cube.js is designed to work with different configuration sources. There
+are two main ways you can set configuration options; via a configuration
+file, commonly known as the `cube.js` file; and environment variables.
+
+There are some areas of overlap between these options, and we recommend
+checking the configuration precedence. In general, defining options in
+your `cube.js` file is the most flexible option because you can define
+the options using JavaScript.
+
+## Configuration Precedence
+
+In Cube.js, environment variables take precedence over values specified
+in `cube.js`. The rationale behind this is simply due to the fact that
+it is generally easier and safer to switch environment variables for
+production than it is to do a new deployment.
```

Comment:
Sounds quite (over?)complicated and "simply", "generally easier", and "safer" are also quite "estimative". I suggest that we describe the rationale without them :)

WDYT about this? (Unsure of the language, just showing the idea.)

```suggestion
in `cube.js`. In production scenarios, it allows to change configuration
by switching environment variables rather than re-deploying the code
 which is generally easier, safer, and faster.
```

Reply:
@igorlukanin way better than what I had, thanks 😄